src/gis/components/Map.js

```diff
@@ -19,19 +109,104 @@ const MapPolygon = props => {
   return <GeoJSON key={uuidv4()} data={geojson} />;
 };
 
+const Location = props => {
+  const map = useMap();
+  const markerRef = useRef(null);
+  const [pinLocation, setPinLocation] = useState();
+  const [boundingBox, setBoundingBox] = useState();
+  const { onPinLocationChange, enableSearch, enableDraw } = props;
+
+  const markerEventHandlers = useMemo(
+    () => ({
+      dragend: () => {
+        const marker = markerRef.current;
+        if (marker != null) {
+          setPinLocation(marker.getLatLng());
+        }
+      },
+    }),
+    []
+  );
+
+  useEffect(() => {
+    if (pinLocation && boundingBox) {
+      onPinLocationChange({
+        pinLocation,
+        boundingBox,
+      });
+    }
+  }, [boundingBox, pinLocation, onPinLocationChange]);
+
+  useEffect(() => {
+    const getZoomData = () => {
+      const southWest = map.getBounds().getSouthWest();
+      const northEast = map.getBounds().getNorthEast();
+      const bbox = [southWest.lng, southWest.lat, northEast.lng, northEast.lat];
+      if (bbox) {
+        setBoundingBox(bbox);
+      }
+    };
+    getZoomData();
+    map.on('zoomend', getZoomData);
+  }, [map]);
+  return (
+    <>
+      {enableSearch && (
+        <LeafletSearch
+          setPinLocation={setPinLocation}
+          setBoundingBox={setBoundingBox}
+        />
+      )}
+
+      {enableDraw && (
+        <LeafletDraw
+          setPinLocation={setPinLocation}
+          setBoundingBox={setBoundingBox}
+        />
+      )}
+
+      {pinLocation && (
+        <Marker
+          draggable
+          ref={markerRef}
+          position={pinLocation}
+          eventHandlers={markerEventHandlers}
+        />
+      )}
+    </>
+  );
+};
+
 const Map = props => {
+  const [map, setMap] = useState();
+
+  useEffect(() => {
+    if (map?.target && props.center) {
+      map.target.flyTo(props.center, 3);
+    }
+  }, [props.center, map]);
+
   return (
     <MapContainer
       zoomDelta={0.5}
       zoomSnap={0.5}
       wheelPxPerZoomLevel={200}
-      {...props}
+      whenReady={setMap}
+      zoom={3}
+      center={[0, 0]}
+      {..._.omit(['center', 'zoom'], props)}
     >
       <TileLayer
-        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        attribution='Data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors Tiles &copy; HOT'
+        url="https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png"
```

Comment:
Question: @paulschreiber why are we using the french url here?

Reply:
This is the "humanitarian" tileset (HOTOSM). Hana preferred this art to the standard art.